lib/ffi-icu/time_formatting.rb

```diff
@@ -65,21 +65,32 @@ def set_attributes(options)
 
       private
 
-      def make_formatter(time_style, date_style, locale, time_zone_str)
-        time_zone = nil
-        d_len = 0
+      def make_formatter(time_style, date_style, locale, time_zone_str, skeleton)
+        time_zone   = nil
+        tz_len      = 0
+        pattern_len = -1
+        pattern_ptr = FFI::MemoryPointer.new(4)
+
         if time_zone_str
           time_zone = UCharPointer.from_string(time_zone_str)
-          d_len = time_zone_str.size
+          tz_len = time_zone_str.size
         else
           Lib.check_error { | error| 
             i_len  = 150
             time_zone = UCharPointer.new(i_len)
-            d_len = Lib.ucal_getDefaultTimeZone(time_zone, i_len, error) 
+            tz_len = Lib.ucal_getDefaultTimeZone(time_zone, i_len, error) 
           }
         end
 
-        ptr = Lib.check_error { | error| Lib.udat_open(time_style, date_style, locale, time_zone, d_len, FFI::MemoryPointer.new(4), -1, error) }
+        if skeleton
+          unless time_style == :pattern && date_style == :pattern
+            raise '`time_style` and `date_style` must be :pattern type when using a skeleton pattern'
+          end
```

Comment:
```suggestion
          date_style = :pattern
          time_style = :pattern
```
It might be a better user experience to just override those two variables, I think if the user is passing skeleton it seems fine to ignore if they set date/time (and plays nice with the fact date_style/time_style defaults to :short, which with the raise would force them to always add `date: => :pattern, :time => :pattern`, without the raise it'd be a bit less typing :))

Also fine if you want to go with the raise logic.

Reply:
Yea I went back and forth with this and ended up going with being explicit. I do think it would be a nicer developer experience when providing a skeleton format to just assume `:pattern` type. I'll make this change.